lib/AdapterRpc.php

```diff
@@ -649,69 +651,134 @@ public function getMemberStatusByUserAndVo($user, $vo)
+            if (empty($resourceCapabilities['value'] ?? [])) {
+                Logger::debug(
+                    self::DEBUG_PREFIX . 'getResourceCapabilities - skipping resource due to empty capabilities.'
+                );
+                continue;
+            }
+            $resourceCapabilities = $resourceCapabilities['value'];
+
+            $resourceGroups = $this->connector->get('resourcesManager', 'getAssignedGroups', [
                 'resource' => $resource['id'],
-                'attributeName' => 'urn:perun:resource:attribute-def:def:capabilities',
-            ])['value'];
+            ]);
 
-            if ($resourceCapabilities === null) {
+            if (empty($resourceGroups)) {
                 continue;
             }
 
             foreach ($resourceGroups as $resourceGroup) {
+                if (($resourceGroup['id'] ?? null) === null) {
+                    Logger::debug(
+                        self::DEBUG_PREFIX . 'getResourceCapabilities - skipping resource group due to missing group ID.'
+                    );
+                    continue;
+                }
                 if (in_array($resourceGroup['id'], $userGroupsIds, true)) {
-                    foreach ($resourceCapabilities as $capability) {
-                        array_push($capabilities, $capability);
-                    }
+                    $capabilities = array_merge($capabilities, $resourceCapabilities);
                     break;
                 }
             }
         }
 
-        return $capabilities;
+        return array_unique($capabilities);
     }
 
-    public function getFacilityCapabilities($entityId, $entityIdAttr = 'perunFacilityAttr_entityID')
+    public function getFacilityCapabilities(string $spEntityId, string $entityIdAttr): array
     {
-        $facility = $this->getFacilityByEntityId($entityId, $entityIdAttr);
+        if (empty($spEntityId)) {
+            Logger::warning(
+                self::DEBUG_PREFIX . 'getFacilityCapabilities - empty spEntityId provided, returning empty list of facility capabilities.'
+            );
+            return [];
+        }
+        $facility = $this->getFacilityByEntityId($spEntityId, $entityIdAttr);
 
         if ($facility === null) {
+            Logger::warning(
+                self::DEBUG_PREFIX . sprintf(
+                    'getFacilityCapabilities - no facility found four EntityID \'%s\', returning empty list of facility capabilities.',
+                    $spEntityId
+                )
+            );
             return [];
         }
 
         $facilityCapabilities = $this->connector->get('attributesManager', 'getAttribute', [
             'facility' => $facility->getId(),
             'attributeName' => 'urn:perun:facility:attribute-def:def:capabilities',
-        ])['value'];
+        ]);
 
-        if (empty($facilityCapabilities)) {
-            $facilityCapabilities = [];
+        if (empty($facilityCapabilities['value'] ?? [])) {
```

Comment:
```suggestion
        if (empty($facilityCapabilities['value'])) {
```
`empty()` handles non-existing keys